presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1451,7 +1445,41 @@ else if (expression instanceof LongLiteral) {
         return analyzedGroupingSets;
     }
 
-    private List<FieldOrExpression> analyzeGroupingColumns(List<Expression> groupingColumns, QuerySpecification node, RelationType tupleDescriptor, AnalysisContext context, List<FieldOrExpression> outputExpressions)
+    private List<Set<Expression>> computeGroupingSetsCrossProduct(List<List<Set<Expression>>> enumeratedGroupingSets, boolean isDistinct)
+    {
+        if (enumeratedGroupingSets.isEmpty()) {
+            return ImmutableList.of();
```

Comment:
I believe this is incorrect. Although I guess this code is dead and can never be executed. But assuming this is live, it should be `ImmutableList.of(ImmutableSet.of())` from context.

I would just remove it and assert the list is not empty if this code is indeed dead.